### PR TITLE
Generate grants.rst

### PR DESCRIPTION
**Pull request by honzajavorek**
Skript, který stáhne informace z https://github.com/pyvec/money a zaarchivuje je do Pyvcí dokumentace jako "zápisy z elektronického hlasování".

### CODE REVIEW

File: _scripts/generate_grants.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import requests
+from jinja2 import Template
+
+
+REACTIONS_API_MEDIA_TYPE = 'application/vnd.github.squirrel-girl-preview'
+GITHUB_TOKEN = os.getenv('GITHUB_TOKEN')
+
+
+url = 'https://api.github.com/repos/pyvec/money/issues'
+res = requests.get(url, headers={'Accept': REACTIONS_API_MEDIA_TYPE,
```
Comment (hroncok) [inline _scripts/generate_grants.py:16]:
```suggestion
res = requests.get(URL, headers={'Accept': REACTIONS_API_MEDIA_TYPE,
```

File: _scripts/generate_grants.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        'title': issue['title'],
+        'description': body,
+        'description_indented': indent(body, '    '),
+        'url': issue['html_url'],
+        'user': {
+            'username': issue['user']['login'],
+            'url': issue['user']['html_url'],
+        },
+        'is_approved': 'approved' in labels,
+        'created_at': to_date(issue['created_at']),
```
Comment (hroncok) [inline _scripts/generate_grants.py:51]: Suggestion: Sort the issues by the date.

**Comment by encukou** [comment]
Should the output should be always approved by a human? IMO, it should.
It looks like here it runs on both `push` and `pull_request`, so if the output changes between adding the pull request and pushing to master, the output won't get a review. Is that right? (I'm not that familiar with GH Actions, to be honest.)

**Comment by hroncok** [comment]
The step is skipped on pull requests, but runs on pushes to master. I.e. currently, there is never a review. If the content goes rogue, we'll fix it with a followup fix. The history is stored in git.

**Comment by hroncok** [comment]
That makes me think that we probably want to add a cronly run.

**Comment by honzajavorek** [comment]
From a lawyer's point of view (but, of course, IANAL), I think it's important that we did the voting, and that it has a record - which is both already fulfilled by the GitHub Issues on the "money" repo.

So the intention is to have an archive of the voting in case anything happens to the "money" repo, and also for our own convenience and searchability. The "money" repo stays as a source of truth, this is only a mirror of the information there. We can gradually improve the quality of the mirror, and if the mirror breaks, we gotta fix it. Hence I wouldn't do any approvals.

Maybe we can add a note that the list is just an automated transcript of the votings and if there's any discrepancy, the "money" repo is the ultimate source of truth.

File: operations/grants.rst.template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+{{ grant.description_indented }}
+
+Výbor o tomto elektronicky hlasoval {{ grant.voted_at.day }}. {{ grant.voted_at.month }}. {{ grant.voted_at.year }}, kdy bylo hlasování uzavřeno s následujícím výsledkem:
+{% for vote in grant.votes %}
+* {{ vote.name }}: {% if vote.content == '+1' -%}
+   ano
+{%- elif vote.content == '-1' -%}
+   ne
+{%- elif vote.content == '' -%}
```
[the remark this answers is not in the document]
Comment (honzajavorek) [reply operations/grants.rst.template:24]: Bug, should be 👀

File: operations/grants.rst.template [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+{% for grant in grants %}
+{{ grant.voted_at.day }}. {{ grant.voted_at.month }}. {{ grant.voted_at.year }} - elektronické hlasování výboru
+--------------------------------------------
+
+Dne {{ grant.filed_at.day }}. {{ grant.filed_at.month }}. {{ grant.filed_at.year }} požádal uživatel `@{{ grant.user.username }} <{{ grant.user.url }}>`_ o grant `{{ grant.title }} <{{ grant.url }}>`_:
+
+.. code-block:: text
+
+{{ grant.description_indented }}
```
[the remark this answers is not in the document]
Comment (honzajavorek) [reply operations/grants.rst.template:16]: That would be very cool!

File: operations/grants.rst.template (hunk 3 above)
[the remark this answers is not in the document]
Comment (hroncok) [reply operations/grants.rst.template:24]: In that case we need to filter out other reactions.

**Comment by honzajavorek** [comment]
I added a few commits. The new batch starts with "use jinja2 filter instead of indent(), yay!" (currently e80b705)

**Comment by honzajavorek** [comment]
Also I was afraid of all the corner cases and growing code complexity, so I added this

image

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -15,6 +15,8 @@ jobs:
       run: python -m pip install -r requirements.txt
     - name: Lint
       run: pylama
+    - name: Test
+      run: pytest
```
Comment (encukou) [inline .github/workflows/main.yml:19]: When things fail, it's better if the logs are verbose:
```suggestion
      run: pytest -v
```

**Comment by honzajavorek** [comment]
Můžem mergnout?

**Comment by hroncok** [comment]
Za mě ano. Ale IANAL.

**Comment by honzajavorek** [comment]
Ajaj, ten soubor se commituje zpátky do gitu, ale tady je protected master 😬 Udělám issue a promyslíme, vyřeším později.

**Pull request merged**